src/data/actions/index.js

```diff
@@ -21,12 +21,28 @@ const createTreeNode = (node, blocks) => (
       node.descendants
         .filter(descendant => blocks[descendant])
         .map(descendant => createTreeNode(blocks[descendant], blocks)),
+  };
+};
+
+
+const getVerticalNodeList = (node, blocks) => {
+  if (node.type == 'vertical') {
+    return [createTreeNode(node, blocks)];
+  } else if (!node.descendants) {
+    return [];
   }
-);
+
+  const reducer = (accumulator, currentValue) => {
+    return accumulator.concat(getVerticalNodeList(blocks[currentValue], blocks));
+  }
+
+  return node.descendants.reduce(reducer, []);
+}
 
 const buildOutlineTree = (blockData) => {
   const rootBlock = blockData.blocks[blockData.root];
-  const outline = createTreeNode(rootBlock, blockData.blocks);
+  let outline = createTreeNode(rootBlock, blockData.blocks);
+  outline['verticalNodeList'] = getVerticalNodeList(rootBlock, blockData.blocks);
```

Comment:
I'd move this out of the `buildOutlineTree` method and into its own method that we can call from the action block